.github/workflows/release.yml

```diff
@@ -1,78 +1,65 @@
-          echo "NEW_VERSION=$(npm --no-git-tag-version version ${{ needs.changelog.outputs.release_type }})" >> $GITHUB_ENV
+        run: npm --no-git-tag-version version ${{ steps.release-changelog.outputs.version }}
 
-      - name: Update changelog unreleased section with new version
-        uses: superfaceai/release-changelog-action@v2
-        with:
-          version: ${{ env.NEW_VERSION }}
-          operation: release
-          format: markdownlint
-
-      - name: Commit CHANGELOG.md and package.json changes and create tag
+      - name: Update repository
         run: |
-          git add "package.json"
-          git add "package-lock.json"
-          git add "CHANGELOG.md"
-          git commit -m "Release ${{ env.NEW_VERSION }}"
-          git tag ${{ env.NEW_VERSION }}
-
-      - name: Run status checks for release commit on temporary branch # Use temporary branch to enable pushing commits to this branch protected by required status checks 
-        uses: CasperWA/push-protected@v2
-        with:
-          token: ${{ secrets.RELEASE_BOT_GITHUB_TOKEN }}
-          branch: main
-          unprotect_reviews: true
-
-      - name: Push changes to repository
-        run: git push origin && git push --tags
+          git config user.name "Open Terms Archive Release Bot"
+          git config user.email "release-bot@opentermsarchive.org"
+          git commit --message="Update changelog and version number" CHANGELOG.md package.json package-lock.json
+          git tag v${{ steps.release-changelog.outputs.version }}
+          git push origin
+          git push --tags
 
-      - name: Read version changelog
-        uses: superfaceai/release-changelog-action@v2
-        id: get-changelog
+      - name: Publish to NPM public repository
+        uses: JS-DevTools/npm-publish@v3
         with:
-          version: ${{ env.NEW_VERSION }}
-          operation: read
+          token: ${{ secrets.NPMJS_ACCESS_TOKEN }}
 
       - name: Create GitHub release
-        uses: softprops/action-gh-release@v1
+        uses: softprops/action-gh-release@v2
         with:
-          tag_name: ${{ env.NEW_VERSION }}
-          body: ${{ steps.get-changelog.outputs.changelog }}
-          token: ${{ secrets.RELEASE_BOT_GITHUB_TOKEN }}
+          tag_name: v${{ steps.release-changelog.outputs.version }}
+          body: ${{ steps.release-changelog.outputs.content }}
+          token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Publish to NPM public repository
-        uses: JS-DevTools/npm-publish@v1
-        with:
-          token: ${{ secrets.NPMJS_ACCESS_TOKEN }}
-          access: public
+  clean-changelog:
+    if: needs.changelog.outputs.release-type == 'no-release'
+    needs: [ changelog ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Update changelog for release
+        uses: OpenTermsArchive/changelog-action/release@v0.4.0
```

Comment:
It's confusing for me to call `changelog-action/release` in the job that clean the changelog.